node/src/node_configurator/node_configurator_standard.rs

```diff
@@ -139,32 +137,33 @@ fn collect_externals_from_multi_config(
 pub fn server_initializer_collected_params<'a>(
     dirs_wrapper: &dyn DirsWrapper,
     args: &[String],
-) -> Result<GatheredParams<'a>, ConfiguratorError> {
+) -> Result<MultiConfig<'a>, ConfiguratorError> {
     let app = app_node();
+    let (config_file_path,
+        config_user_specified,
+        data_directory,
+        data_directory_specified,
+        real_user,
+        real_user_specified,
+        pre_orientation_args) = determine_user_specific_data(dirs_wrapper, &app, args)?;
+    let mut full_multi_config_vec: Vec<Box<dyn VirtualCommandLine>> = vec![
+            Box::new(EnvironmentVcl::new(&app)),
+            pre_orientation_args
+        ];
 
-    let (config_file_path, user_specified, data_directory, real_user) =
-        determine_fundamentals(dirs_wrapper, &app, args)?;
-
-    let config_file_vcl = match ConfigFileVcl::new(&config_file_path, user_specified) {
-        Ok(cfv) => Box::new(cfv),
-        Err(e) => return Err(ConfiguratorError::required("config-file", &e.to_string())),
+    let mut fill_specified_or_unspecified_box = |key: String, value: String, specified: bool | {
```

Comment:
Again, nobody really cares what data structure serves as the medium (whether a Box or a Vec), he can read it from the code easily. To express the purpose is always helpful, though. 

Maybe: `sort_out_by_whether_user_specified`.